[stable/concourse] Add config for the port name

austinbv opened the pull request:
This allows for a bit of automoation when it comes to spinning up aws lbs

*****************************************************************************************

Thank you for contributing to kubernetes/charts. Before you submit this PR we'd like to
make sure you are aware of our technical requirements and best practices:

* https://github.com/kubernetes/charts/blob/master/CONTRIBUTING.md#technical-requirements
* https://github.com/kubernetes/helm/tree/master/docs/chart_best_practices

For a quick overview across what we will look at reviewing your PR, please read
our review guidelines:

* https://github.com/kubernetes/charts/blob/master/REVIEW_GUIDELINES.md

Following our best practices right from the start will accelerate the review process and
help get your PR merged quicker.

When updates to your PR are requested, please add new commits and do not squash the 
history. This will make it easier to identify new changes. The PR will be squashed 
anyways when it is merged. Thanks.

*****************************************************************************************

austinbv committed:
[stable/concourse] Add config for the port name

This allows for a bit of automoation when it comes to spinning up aws lbs

austinbv commented:
I signed the agreement for the foundation now

austinbv commented:
Looks like circleci failed for some unknown reason. We should try and rerun it.

Thanks for looking at my PR!

austinbv commented:
/assign @foxish

william-tran commented on stable/concourse/README.md:
```diff
@@ -77,7 +77,9 @@ The following tables lists the configurable parameters of the Concourse chart an
 | `concourse.workerKeyPub` | Concourse Worker Public Key | *See [#ssh-keys](#ssh-keys)* |
+| `concourse.atcPortName` | Name of Concourse ATC listen port | `atc` |
 | `concourse.atcPort` | Concourse ATC listen port | `8080` |
+| `concourse.tsaPortName` | Name of Concourse TSA listen port | `name` |
```
`tsa`

william-tran commented on stable/concourse/values.yaml:
```diff
@@ -139,11 +139,21 @@ concourse:
   encryptionKey:
 
+  ## Name of ATC listen port.
+  ## Name of ref: https://concourse.ci/architecture.html
```
Name of ref?

william-tran reviewed (changes requested):
Could you also provide more details on the use case?

Commit:
[stable/wordpress] Release 0.8.1 (#3353)

* wordpress: update to `bitnami/wordpress:4.9.2-r0`

* wordpress: bump chart appVersion to `4.9.2`

* wordpress: bump chart version to `0.8.1`

Commit:
[stable/phabricator] Release 0.5.7 (#3355)

* phabricator: update to `bitnami/phabricator:2018.2.0-r0`

* phabricator: bump chart appVersion to `2018.2.0`

* phabricator: bump chart version to `0.5.7`

austinbv commented:
@william-tran Sure happy to.  We have some pretty significant build automation and a strong convention that we name all parts of our PODs the same across a few different systems. In this case we rely on the `port.name` to match the `servicename`. We talked about either changing our build automation or attempting this PR and figured the PR may be nicer if anyone else is in our situation. Removing hard coding while applying sensible defaults - we thought - increased portability and utility of the chart.

Hopefully this context helps. This small change should allow us to get concourse up and running!

william-tran commented:
@austinbv I don't see the port name being configurable in other charts so I'm hesitant to give my thumbs up for this unless there are others that can see this feature being useful for a number of users. I'm still pretty new to k8s and chart maintenance, so I'll defer judgement to someone else with more experience. @viglesiasce @mattfarina ?

braintreeps committed:
Provide a custom port name for atc

mattfarina commented:
/cc @frodenas @viglesiasce @nexeck since this is also a chart you maintain

cc @unguiculus do you have any guidance on the practices here?

There are cases where others have named ports after service names. For example, see [sentry](https://github.com/kubernetes/charts/blob/cb0060835763bcdd68fb080ab4799f36a1033fa1/stable/sentry/templates/service.yaml). In that case it didn't need to be an extra property but could be derived from the service name automatically.

mattfarina commented on stable/concourse/templates/web-svc.yaml:
```diff
@@ -18,13 +18,13 @@ metadata:
   type: {{ .Values.web.service.type }}
   ports:
-    - name: atc
+    - name: {{ .Values.concourse.atcPortName }}
```
If you're looking to name the ports in a similar or the same manner to the service would it make sense to base it on the service name without an extra property? Something like what the [sentry chart](https://github.com/kubernetes/charts/blob/cb0060835763bcdd68fb080ab4799f36a1033fa1/stable/sentry/templates/service.yaml#L16) does?

austinbv replied:
Nice I can update the PR to do something like sentry

mattfarina reviewed (commented):
The chart is failing CI because the chart version also needs to be incremented.

Commit:
Fluentd-Cloudwatch Changes (#3229)

* Fluentd-Cloudwatch Changes

- Add support for awsRoles vai kube2iam role
- Add rbac create support
- Standardize helpers with other charts

* Fix the lint errors

* Forgot to update the secrets

* Address Code Review Comments
